File: src/main/java/cz/vhromada/catalog/rest/controller/LanguageController.java
package cz.vhromada.catalog.rest.controller;

import java.util.Arrays;
import java.util.List;

import cz.vhromada.common.Language;
import cz.vhromada.validation.result.Result;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * A class represents controller for language.
 *
 * @author Vladimir Hromada
 */
@RestController("languageController")
@RequestMapping("/catalog/languages")
public class LanguageController extends AbstractCatalogController {

    /**
     * Returns list of languages.
     *
     * @return result with list of languages
     */
    @GetMapping
    public ResponseEntity<Result<List<Language>>> getLanguages() {
        return processResult(Result.of(Arrays.asList(Language.values())));
    }

    /**
     * Returns list of subtitles.
     *
     * @return result with list of subtitles
     */
    @GetMapping("/subtitles")
    public ResponseEntity<Result<List<Language>>> getSubtitles() {
        return processResult(Result.of(Arrays.asList(Language.CZ, Language.EN)));
    }

}

File: src/main/kotlin/cz/vhromada/catalog/rest/controller/LanguageController.kt
package cz.vhromada.catalog.rest.controller

import cz.vhromada.common.Language
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * A class represents controller for language.
 *
 * @author Vladimir Hromada
 */
@RestController("languageController")
@RequestMapping("/catalog/languages")
class LanguageController {

    /**
     * Returns list of languages.
     *
     * @return list of languages
     */
    @GetMapping
    fun getLanguages(): List<Language> {
        return Language.values().asList()
    }

    /**
     * Returns list of subtitles.
     *
     * @return list of subtitles
     */
    @GetMapping("/subtitles")
    fun getSubtitles(): List<Language> {
        return listOf(Language.CZ, Language.EN)
    }

}
